### PR TITLE
Keep objects of shoot managed resources during migration

### PR DESCRIPTION
**Pull request by stoyanr**
**How to categorize this PR?**

/area control-plane-migration
/kind bug
/priority normal

**What this PR does / why we need it**:
Keep objects of shoot managed resources during migration:

* Vendor gardener/gardener@v1.12.5 (needed for the `KeepManagedResourceObjects` utility function)
* Set `keepObjects` to true for shoot managed resources during migration before deleting them, so that the actual shoot objects are not deleted.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
All extensions should take care about their managed resources during the migration. Unless there is a special reason not to do so, all managed resources should be deleted after setting `keepObjects` to true, so that the actual shoot objects are not deleted.

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

**Review by plkokanov — CHANGES_REQUESTED** [review]
I think you should also change the golang version in the `Dockerfile` to
`FROM golang:1.15.3 AS builder`

Otherwise lgtm.

**Comment by stoyanr** [comment]
@plkokanov thanks, I will fix this. I am still getting the following error in `check-generate`:
```
>> make revendor
make revendor needs to be run:
 M vendor/modules.txt
make: *** [Makefile:84: check-generate] Error 1
```
I can't reproduce it locally and I don't know what to do. I ran `make revendor` a few times already. Any ideas?

**Comment by stoyanr** [comment]
@MartinWeindel @timuthy Could you please help? I tried a number of things to fix the above error, but no luck. `make check-generate` simply passes when I run it locally:
```
$ make check-generate
> Generate / Vendor Check
>> make generate
>> make revendor
```

**Comment by stoyanr** [comment]
I logged into the container using `fly` and ran `make generate`. The differences in `vendor/modules.txt` were:

* All `## explicit` comments were missing.
* The following comments section at the end was missing:
```
-# k8s.io/api => k8s.io/api v0.18.8
-# k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.18.8
-# k8s.io/apimachinery => k8s.io/apimachinery v0.18.8
-# k8s.io/apiserver => k8s.io/apiserver v0.18.8
-# k8s.io/client-go => k8s.io/client-go v0.18.8
-# k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.18.8
-# k8s.io/code-generator => k8s.io/code-generator v0.18.8
-# k8s.io/component-base => k8s.io/component-base v0.18.8
-# k8s.io/helm => k8s.io/helm v2.13.1+incompatible
-# k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.18.8
```

I have no idea how to configure `make revendor` to generate this "correctly", so I fixed `vendor/modules.txt` manually. However, this is a real pain, @MartinWeindel @timuthy could you please take a look and fix this?

**Comment by ialidzhikov** [comment]
@stoyanr , the issue is that the branch.cfg was configured to run with `golang:1.13.7`. I now updated it to latest golang with 8d84a19536d3ffab617f6f57652f70b61647799c. Please vendor locally properly and push without any hacks.

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ############# builder
-FROM golang:1.14.9 AS builder
+FROM golang:1.15.3 AS builder
```
Comment (ialidzhikov) [inline Dockerfile:2]:
```suggestion
FROM golang:1.15.5 AS builder
```

Comment (stoyanr) [reply Dockerfile]: Done

**Comment by stoyanr** [comment]
@ialidzhikov 
> the issue is that the branch.cfg was configured to run with `golang:1.13.7`. I now updated it to latest golang with [8d84a19](https://github.com/gardener/gardener-extension-shoot-cert-service/commit/8d84a19536d3ffab617f6f57652f70b61647799c). Please vendor locally properly and push without any hacks.
Did it but the verify job still fails with the above error.

**Comment by ialidzhikov** [comment]
Here you go - ref https://concourse.ci.gardener.cloud/teams/gardener/pipelines/gardener-extension-shoot-cert-service-master/jobs/master-pull-request-job/builds/18. I just takes time until the changes in `branch.cfg` are applied.

**Review by ialidzhikov — CHANGES_REQUESTED** [review]
Can you add `SKIP_FETCH_TOOLS=1` to `make test` and `make test-cov`?

**Comment by stoyanr** [comment]
> Can you add `SKIP_FETCH_TOOLS=1` to `make test` and `make test-cov`?

Done.

**Review by MartinWeindel — APPROVED** [review]
/lgtm
Thanks!

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**